Sources/xcproj/PBXProj.swift

```diff
@@ -73,33 +74,65 @@ final public class PBXProj: Decodable {
-            case let object as PBXHeadersBuildPhase: headersBuildPhases.append(object)
-            case let object as PBXNativeTarget: nativeTargets.append(object)
-            case let object as PBXFileReference: fileReferences.append(object)
-            case let object as PBXProject: projects.append(object)
-            case let object as XCVersionGroup: versionGroups.append(object)
-            case let object as PBXReferenceProxy: referenceProxies.append(object)
+                legacyTargets.append(object, reference: reference)
+            case let object as PBXContainerItemProxy: containerItemProxies.append(object, reference: reference)
+            case let object as PBXCopyFilesBuildPhase: copyFilesBuildPhases.append(object, reference: reference)
+            case let object as PBXGroup: groups.append(object, reference: reference)
+            case let object as XCConfigurationList: configurationLists.append(object, reference: reference)
+            case let object as XCBuildConfiguration: buildConfigurations.append(object, reference: reference)
+            case let object as PBXVariantGroup: variantGroups.append(object, reference: reference)
+            case let object as PBXTargetDependency: targetDependencies.append(object, reference: reference)
+            case let object as PBXSourcesBuildPhase: sourcesBuildPhases.append(object, reference: reference)
+            case let object as PBXShellScriptBuildPhase: shellScriptBuildPhases.append(object, reference: reference)
+            case let object as PBXResourcesBuildPhase: resourcesBuildPhases.append(object, reference: reference)
+            case let object as PBXFrameworksBuildPhase: frameworksBuildPhases.append(object, reference: reference)
+            case let object as PBXHeadersBuildPhase: headersBuildPhases.append(object, reference: reference)
+            case let object as PBXNativeTarget: nativeTargets.append(object, reference: reference)
+            case let object as PBXFileReference: fileReferences.append(object, reference: reference)
+            case let object as PBXProject: projects.append(object, reference: reference)
+            case let object as XCVersionGroup: versionGroups.append(object, reference: reference)
+            case let object as PBXReferenceProxy: referenceProxies.append(object, reference: reference)
             default: fatalError("Unhandled PBXObject type for \(object), this is likely a bug / todo")
             }
         }
+        
+        /// Generates a deterministic reference from an object type and identifier.
+        /// It ensures that the generated reference doesn't collide with any existing one.
+        ///
+        /// - Parameters:
+        ///   - element: object type.
+        ///   - id: object identifier (e.g. path or name)
+        /// - Returns: reference.
+        public func generateReference<T: PBXObject>(_ element: T.Type, _ id: String) -> String {
```

Comment:
Will the old generation function get deleted?

Reply:
Yes it will, I'll remove it.

Reply:
Removed